torch/_inductor/scheduler.py

```
@@ -1276,6 +1276,16 @@ def _get_atomic_add_buffers(self) -> OrderedSet[str]:
                     )
         return buffers_store_as_atomic_add
 
+    @cache_on_self
+    def has_side_effects(self) -> bool:
+        if (
+            self._body is not None
+            and hasattr(self._body, "has_op")
+            and self._body.has_op("device_assert_async")
```

scheduler node _body should always not be None. We should clear what the invariants of the codebase are, otherwise it makes it confusing to read. If `_body` is None and it shouldnt be - we do actually want to surface the error. 

So I would be careful about this type of "defensive coding"

---

I completely agree with your point. Removed the none condition.